include/sway/tree/arrange.h

```diff
@@ -0,0 +1,20 @@
+#ifndef _SWAY_ARRANGE_H
+#define _SWAY_ARRANGE_H
+
+struct sway_container;
+
+void arrange_windows(struct sway_container *container);
+
+// Determine the root container's geometry, then iterate to everything below
+void arrange_root(void);
+
+// Determine the output's geometry, then iterate to everything below
+void arrange_output(struct sway_container *output);
```

Comment:
I'm not a fan of these "I take a generic `struct sway_container` as input but I only work on output/workspace". Basically these functions could be typed and take a `sway_output`/`sway_workspace`.

Reply:
`arrange_children_of()` would still need to accept a `sway_container`, and that means there'd be inconsistency in how you call these functions.

Additionally, there'd be nothing stopping you from accidentally passing the wrong pointer for that container type to the function. For example, if you had a C_CONTAINER, assumed it was a workspace by mistake and passed its sway_workspace pointer to `arrange_workspace()`. That wouldn't be caught during compilation, and `arrange_workspace()` wouldn't be able to assert that this has not happened.

Reply:
I don't see the value of having 4 functions taking "generic containers, but not really" instead of one.

When you access a sway_workspace field you must _always_ have an assert just above checking the type (or you already have the specialized type).

Reply:
¯\\\_(ツ)\_/¯

Sort it out between yourselves

Reply:
@RyanDwyer I just don't get why there's a need for these per-type functions instead of a generic one (which is here anyway). Can you explain me why you made this change?

Reply:
The previous function had a lot of complexity which made it hard to understand. It was recursive, used different logic per container type, and some of the switch statements used return while others use break. Separating them into different functions improves readability because the part that is recursive is isolated to a smaller area of code.

In hindsight, I wouldn't have included the new `arrange_windows()` function at all. It's not called anywhere, and it's not a very appropriate name because it can also arrange outputs and workspaces.

Reply:
>Separating them into different functions improves readability because the part that is recursive is isolated to a smaller area of code.

Sure, but these functions could be static.

>It's not called anywhere, and it's not a very appropriate name because it can also arrange outputs and workspaces.

Can you make a PR to remove it then?